### ISSUE TITLE
Meson incorrectly detects MinGW environment when running under MSYS2

### ISSUE THREAD

**Issue by dtromb** [mesonbuild/meson#12772]
On a fresh install of MSYS2, in the default MSYS2 shell, after running `pacman -S meson`, meson emits the error:

> ERROR: This python3 seems to be msys/python on MSYS2 Windows, but you are in a MinGW environment
> ERROR: Please install and use mingw-w64-x86_64-python3 and/or mingw-w64-x86_64-meson with Pacman
> ERROR: This python3 seems to be msys/python on MSYS2 Windows, but you are in a MinGW environment
> ERROR: Please install and use mingw-w64-x86_64-python3 and/or mingw-w64-x86_64-meson with Pacman

The shell is not in the MinGW environment, it is MSYS2.   If we are meant to use the mingw python when running meson under MSYS2, the error message should be changed to make sense for that case and also there needs to be some way to get a working python/meson into the MSYS2 environments (the recommended install command installs into the mingw environment, which I am not using.)

If on the other hand meson is incorrectly detecting the mingw environment present, that also needs a fix.

In short:  meson should run properly after some set of install commands when run as 'meson' - this is breaking many builds on MSYS2.  

I understand that this project does not have control over the MSYS2 repository, but it is meson that is exiting in error and giving a non-functional workaround here.  

Finally, if running meson under the MSYS2 shell is not supported in any way is not supported and we need to use the mingw shell, that also needs to be made explicit and the error message changed.

**Comment by dtromb** [comment]
mesonmain.py:253

    if sys.platform == 'cygwin' and os.environ.get('MSYSTEM', '') not in ['MSYS', '']:
    
The actual value of $MSYSTEM in my shell is `UCRT64`.  

Likely related info here: https://www.msys2.org/news/#2022-10-29-changing-the-default-environment-from-mingw64-to-ucrt64

**Comment by lazka** [comment]
The detection is still correct, every non-MSYS environment is a "mingw" environment. The thing that is out of date is the error message suggesting "mingw-w64-x86_64-python3" which installs it for MINGW64, but with UCRT64 you need "mingw-w64-ucrt-x86_64-python" instead.

Maybe instead of the package names we could point to https://packages.msys2.org/base/mingw-w64-python ?
